diff --git a/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift b/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift
index f4f2ddd..7d89a3c 100644
--- a/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift
+++ b/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift
@@ -120,7 +120,7 @@ extension ConcurrentFunctionDefinition: ViewDefinitionFunctionElement {
   public typealias ViewType = FirstArgType
 }
 
-extension UIView: @MainActor AnyArgument {
+extension UIView: AnyArgument {
   public static func getDynamicType() -> AnyDynamicType {
     return DynamicViewType(innerType: Self.self)
   }
